components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue

```diff
@@ -1,46 +1,167 @@
-    <GalleryItemActionSlides
-      v-if="Number(nft.price)"
-      ref="actionRef"
-      :active="active">
-      <template #action>
-        <NeoButton
-          :label="label"
-          size="large"
-          fixed-width
-          variant="k-accent"
-          no-shadow
-          @click.native="toggleActive" />
-      </template>
-
-      <template #content>
-        <div class="has-text-centered">
-          Buy NFT on
-          <span class="has-text-weight-bold is-uppercase">{{ urlPrefix }}</span>
-        </div>
-      </template>
-    </GalleryItemActionSlides>
-    <div v-else>Not Listed</div>
-  </GalleryItemPriceSection>
+  <div>
+    <Loader v-model="isLoading" :status="status" />
+    <GalleryItemPriceSection v-if="nftPrice" title="Price" :price="nftPrice">
+      <GalleryItemActionSlides
+        v-if="Number(nftPrice)"
+        ref="actionRef"
+        :active="active">
+        <template #action>
+          <NeoButton
+            :label="label"
+            size="large"
+            fixed-width
+            variant="k-accent"
+            :disabled="disabled"
+            no-shadow
+            @click.native="onClick" />
+        </template>
+
+        <template #content>
+          <div class="has-text-centered">
+            Buy NFT on
+            <span class="has-text-weight-bold is-uppercase">{{
+              urlPrefix
+            }}</span>
+          </div>
+        </template>
+      </GalleryItemActionSlides>
+      <div v-else>Not Listed</div>
+    </GalleryItemPriceSection>
+  </div>
 </template>
 
 <script setup lang="ts">
 import { NeoButton } from '@kodadot1/brick'
 import { onClickOutside } from '@vueuse/core'
+import { notificationTypes, showNotification } from '@/utils/notification'
 
-import { useGalleryItem } from '../../useGalleryItem'
 import GalleryItemPriceSection from '../GalleryItemActionSection.vue'
 import GalleryItemActionSlides from '../GalleryItemActionSlides.vue'
+import { tokenIdToRoute } from '@/components/unique/utils'
+import { getApiCall } from '@/utils/gallery/abstractCalls'
+import { getKusamaAssetId } from '@/utils/api/bsx/query'
+import { JustInteraction, createInteraction } from '@kodadot1/minimark'
+import { somePercentFromTX } from '@/utils/support'
+import nftByIdMinimal from '@/queries/rmrk/subsquid/nftByIdMinimal.graphql'
+
+const props = defineProps<{
+  nftId: string
+  currentOwner: string
+  collectionId: string
+  nftPrice: string
+}>()
 
-const { urlPrefix } = usePrefix()
-const { nft } = useGalleryItem()
+const { urlPrefix, client } = usePrefix()
+const { accountId } = useAuth()
+const { $store, $apollo } = useNuxtApp()
+const { apiInstance } = useApi()
+const ACTION = 'BUY'
+
+const { howAboutToExecute, initTransactionLoader, isLoading, status } =
+  useMetaTransaction()
 
 const active = ref(false)
 const label = computed(() => (active.value ? 'Confirm' : 'Buy'))
 
-function toggleActive() {
-  active.value = !active.value
+const balance = computed<string>(() => {
```

Comment:
@roiLeo 
I'm not sure that this is the correct way to do this

more generally, I would like to know please....
lets say i'm looking to buy this item:
https://kodadot.xyz/bsx/gallery/2548799063-138
which cost 0.49 ksm on the BSX network
should my wallet contain 0.49 ksm or the BSX equivalent?

Reply:
On bsx idk, it might have some transaction fees (but it's really minimal) whereas on RMRK we have integrated platform fees (around .3cent if i recall correctly `somePercentFromTX`)

Reply:
the ```somePercentFromTx``` is used in ```getTranasactionParams``` function
I think this balance calaulation is correct, it does the same as galleryItem in bsx an in rmrek were doing respectively